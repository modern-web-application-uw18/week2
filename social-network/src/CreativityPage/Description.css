.ArticleDescription {
    color: rgba(0,0,0,.54);
    fill: rgba(0,0,0,.54);
    font-size: 14px;
    position: relative;
    padding-top: 10px;
}