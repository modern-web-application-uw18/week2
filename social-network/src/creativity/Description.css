.Description {
    text-align: left;
}

.Article-header {
    max-width: 220px;
    font-weight: 700;
    font-size: 1em;
    color: black;
    vertical-align: "top";
}
      
.Article-description {
    vertical-align: "middle";
    color: gray;
    font-size: 0.75em;
    font-weight: 500;
    padding-top: 10px;
}

