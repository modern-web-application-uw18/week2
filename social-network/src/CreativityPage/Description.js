import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Description.css'

class Description extends Component {

    render() {
        const articleDescription = this.props.description;
        return (
            <div className="ArticleDescription">{articleDescription}</div>
        );
    }
}

Description.propTypes = {
    description: PropTypes.string
};

Description.defaultProps = {
    description: ''
};

export default Description;