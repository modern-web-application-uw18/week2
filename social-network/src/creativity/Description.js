import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Description.css';

class Description extends Component {

    render() {

        return (
            <div className="Description">
                <div className="Article-header">{this.props.title}</div>
                <div className="Article-description">{this.props.description}</div>
            </div>
        );
    }
}


Description.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
};


Description.defaultProps = {
    title: 'No title :(',
    description: 'No description :('
};

export default Description;