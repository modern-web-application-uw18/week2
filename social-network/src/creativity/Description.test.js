import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Description from './Description';

describe('ArticleInfo', () => {
    it('renders without crashing', () => {
        const articleInfo = document.createElement('ArticleInfo');
        ReactDOM.render(<Description
            title="Captain Jean-Luc Picard"
            description="Musings about the Star Trek Next Generation series" />, articleInfo);
        ReactDOM.unmountComponentAtNode(articleInfo);
    });

    it('should render with title, description', () => {
        const component = TestRenderer.create(
            <Description title="Resume Magic" description="How to craft the perfect resume" />);
            expect(component.toJSON()).toMatchSnapshot();
    });
});