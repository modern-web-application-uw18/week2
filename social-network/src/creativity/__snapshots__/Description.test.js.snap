// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleInfo should render with title, description 1`] = `
<div
  className="Description"
>
  <div
    className="Article-header"
  >
    Resume Magic
  </div>
  <div
    className="Article-description"
  >
    How to craft the perfect resume
  </div>
</div>
`;
